### ISSUE TITLE
Scrape new taps/targets from GitHub, especially ones built on SDK

### ISSUE THREAD

**Issue by DouweM** [meltano/hub#699]

**Comment by tayloramurphy** [comment]
@pnadolny13 let's chat about this next week. We'll need to get some regular data processing up for the Hub generally.

**Comment by pnadolny13** [comment]
@DouweM @tayloramurphy yeah I like the idea of having a workflow that automates finding the taps/targets and collecting some minimal info but leaves it up to the reviewer to take it across the finish line. At a minimum it should open an issue so we can track all plugins we need to add but we can probably have it open a draft PR with the basic info. Even if it only opens an issue we will at least have a way to track our backlog that need to get added. Something potentially like this:

Automated Workflow:
- scrape github for taps/target
- diff against whats already on the hub. Just for those that dont exist, not trying to diff setting or anything like that right now.
- diff against existing issue/PR draft to not open multiple for the same variant.
- build a minimum [definition json](https://docs.meltano.com/reference/plugin-definition-syntax). More detailed thoughts below.
- open draft PR and tag reviewer, probably me

Review Process:
- check for general accuracy
- open up repo and look for capabilities
- add logo image
- add settings/usage docs/etc. as needed
- add maintenance status
- convert

A challenge might be differentiating between when a tap is brand new and not ready to be listed vs ready. Maybe as part of the manual process we assign the issue to a backlog or something if its not ready but we'd need to periodically groom it and look for plugins that have matured enough to be added.


### Automated plugin definition 

Given [definition json](https://docs.meltano.com/reference/plugin-definition-syntax) there are some things we can automatically define and others that have to be manual.

- ✅ name
- ✅ variant
- ✅ namespace
- ✅ label
- ✅ repo
- ❓ capabilities - we can give it a best guess but would need to be reviewed manually. Looking for certain words in the README like "state.json" or "--state"
- ❓ pip_url - we can assign the git url and the reviewer can look for a pypi alternative.
- ✅ maintenance_status - assume all are active and let reviewer judge or make a call on when the most recent commit needs to be <=6 months or similar.
- ❓  keywords - we can try to detect if its meltano_sdk from the README, expecting a manual review.
- ❌ settings
- ❌ settings_group_validation

OPTIONAL
- ❌ domain_url
- ❌ docs
- ❌  settings_preamble (optional)
- ❌  next_steps (optional)
- ❌  usage (optional)
- ❌ requires (optional anyways)
- ❌ commands (NA)

**Comment by pnadolny13** [comment]
@tayloramurphy I was tinkering with this over the weekend and I'm thinking about creating a `target-meltanohub` that manages this. We had a similar pattern at my previous job where we built smart custom targets that did workflows like this. A major benefit being that it plugs right into our current Meltano deployment. The idea would be:

- dbt models to compile plugin definitions based on our github search tap
- create a view of plugin definitions that arent on meltanohub yet. Maybe using Derek's audit table pattern https://youtu.be/RXo-lIQtwB8?t=1789
- new job syncing tap-snowflake -> target-meltanohub
- the target will clone the hub, then for each record:
  - create new branch
  - add the new plugin variant
  - check if the maintainer exists or add it
  - commit changes
  - push to github
  - open PR (maybe draft PRs)
- the target would use a PR naming convention so it can query for open PRs and not open duplicates. Something like `[MeltanoHub Bot] Add Plugin MeltanoLabs/tap-github`
- for the case where a reviewer decides that the plugin isnt ready to be listed on the hub, maybe its still in development, we close the PR after review. The target wont open a duplicate but it can have logic to create a new PR if its been some amount of time since the last one, 1-3 months later maybe? Then we can re-review and evaluate it again. Something like `[MeltanoHub Bot] Add Plugin MeltanoLabs/tap-github (Attempt 2)`

What do you think about that idea?

**Comment by tayloramurphy** [comment]
> create a view of plugin definitions that arent on meltanohub yet

this could be a fun chance to play with https://github.com/datafold/data-diff if it makes sense.

Otherwise, your plan looks good!

**Comment by pnadolny13** [comment]
@rwfeather do you have any thoughts about my proposed process in https://github.com/meltano/hub/issues/699#issuecomment-1215071376 or about the automation in general?

**Comment by rwfeather** [comment]
> @rwfeather do you have any thoughts about my proposed process in [#699 (comment)](https://github.com/meltano/hub/issues/699#issuecomment-1215071376) or about the automation in general?

@pnadolny13 I like your proposal for automating this way. Scraping on a regular basis then comparing the result to our list of open/closed PRs and existing definitions makes sense.
